src/widgets/settingspages/GeneralPage.hpp

```diff
@@ -93,6 +95,36 @@ class SettingsLayout : public QVBoxLayout
         return button;
     }
 
+    template <typename T>
+    ColorButton *addColorButton(const QString &text, const QColor &color,
+                                pajlada::Settings::Setting<T> &setting)
+    {
+        auto colorButton = new ColorButton(color);
+        auto layout = new QHBoxLayout();
+        auto label = new QLabel(text + ":");
+        layout->addWidget(label);
+        layout->addStretch(1);
+        layout->addWidget(colorButton);
+        this->addLayout(layout);
+        QObject::connect(
+            colorButton, &ColorButton::clicked, [setting, colorButton]() {
```

Comment:
nit: `ColorButton::color()` could be used to obtain the color instead of also passing `setting` to the lambda